### PR TITLE
update: TGI  Integrated

### PR DESCRIPTION
**Pull request by oat-mirror**
guidance +  tgi generate API

### CODE REVIEW

File: pyproject.toml
Comment (edwardzjl) [inline pyproject.toml:1]: Adding this and the `requirements.txt` file will break the 'single source of truth' rule, I suggest make minimal change on the `setup.py` file instead.

File: guidance/llms/_model.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import os
+
+from loguru import logger
```
Comment (edwardzjl) [inline guidance/llms/_model.py]: I don't think using loguru in a library is a good idea, especially the upstream repo does not rely on loguru. Instead we might just use the plain `logging.getLogger` function, or refer to how the upstream does.

(Please check <https://github.com/Delgan/loguru/issues/349#issuecomment-707267009>)

File: guidance/llms/_model.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+from ._llm import LLM
+
+from text_generation import AsyncClient
+from transformers.configuration_utils import PretrainedConfig
+
+endpoint_url = os.getenv("ENDPOINTS_URL", "http://127.0.0.1:8080/")
```
Comment (edwardzjl) [inline guidance/llms/_model.py]: 'TGI_ENDPOINT_URL'?

File: pyproject.toml
Comment (oat-mirror) [reply pyproject.toml:1]: fixed.

File: guidance/llms/_model.py (hunk 2 above)
Comment (oat-mirror) [reply guidance/llms/_model.py]: fixed.

File: guidance/llms/_model.py (hunk 1 above)
Comment (oat-mirror) [reply guidance/llms/_model.py]: Builtin logging has been used instead of loguru..

**Pull request merged**